Rewrite `tea.backend.SScriptGlobalMap`

CaptainBaldi opened the pull request:
1. Allows for the use of `SScript.globalVariables` inside of SScript instances again, by instead of using an abstract, `SScriptGlobalMap` now uses an internal `StringMap`.
2. Added other `StringMap` functions (`keys`, `iterator`, etc).

I'm not too familiar with how `iterator`/`keyValueIterator` work, so if there are any issues please let me know.

TheWorldMachinima commented:
> 1. Allows for the use of `SScript.globalVariables` inside of SScript instances again, by instead of using an abstract, `SScriptGlobalMap` now uses an internal `StringMap`.
> 2. Added other `StringMap` functions (`keys`, `iterator`, etc).
> 
> I'm not too familiar with how `iterator`/`keyValueIterator` work, so if there are any issues please let me know.

I haven't thought about using global variables when I rewrote it as an abstract. Today I will rewrite it as an IMap, if I fail to do so I will reopen this PR and merge it.

CaptainBaldi commented:
> > 1. Allows for the use of `SScript.globalVariables` inside of SScript instances again, by instead of using an abstract, `SScriptGlobalMap` now uses an internal `StringMap`.
> > 2. Added other `StringMap` functions (`keys`, `iterator`, etc).
> > 
> > I'm not too familiar with how `iterator`/`keyValueIterator` work, so if there are any issues please let me know.
> 
> I haven't thought about using global variables in runtime when I rewrote it as an abstract. Today I will rewrite it as an IMap, if I fail to do so I will reopen this PR and merge it.

👍
